rr_platform/src/camera/CameraGeometry.cpp

```diff
@@ -0,0 +1,146 @@
+#include "CameraGeometry.h"
+
+#include <cmath>
+#include <tf/transform_listener.h>
+#include <tf/transform_datatypes.h>
+
+#include "angle_utils.hpp"
+
+
+namespace rr {
+
+void CameraGeometry::CameraInfoCallback(const sensor_msgs::CameraInfoConstPtr& msg) {
+  image_size_rows_ = msg->height;
+  image_size_cols_ = msg->width;
+  double fx = msg->P[0];  // horizontal focal length of rectified image, in px
+  double fy = msg->P[5];  // vertical focal length
+  cam_fov_x_ = 2 * atan2(image_size_cols_, 2*fx);
+  cam_fov_y_ = 2 * atan2(image_size_rows_, 2*fy);
+  received_camera_info_ = true;
+}
+
+bool CameraGeometry::LoadFOV(ros::NodeHandle &nh, const std::string &camera_info_topic,
+                             const double timeout)
+{
+  received_camera_info_ = false;
+
+  // temporary camera info subscriber is destroyed after one message
+  auto cam_info_sub = nh.subscribe(camera_info_topic, 1, &CameraGeometry::CameraInfoCallback, this);
+
+  ros::Time t_start = ros::Time::now();
+  while (!received_camera_info_) {
+    if ((ros::Time::now() - t_start).toSec() > timeout) {
+      ROS_WARN("[CameraGeometry] setting camera FOV from camera_info timed out");
+      break;
+    }
+
+    ros::spinOnce();
+    ros::Duration(0.1).sleep();
+  }
+
+  return received_camera_info_;
+}
+
+bool CameraGeometry::LoadCameraPose(const std::string &camera_link_name, const double timeout) {
+  geometry_msgs::PoseStamped ps_src_cam;  // source pose, origin
+  geometry_msgs::PoseStamped ps_dst_base;  // destination pose, camera frame origin from base_footprint
+
+  ps_src_cam.pose.orientation.w = 1.0;  // set orientation in x-axis direction
+  ps_src_cam.header.frame_id = camera_link_name;
+
+  tf::TransformListener listener;
+  bool success = false;
+  while (!success) {
+    try {
+      listener.waitForTransform("base_footprint", camera_link_name, ros::Time(0), ros::Duration(timeout));
```

Comment:
This never returns false. Either remove the loop or implement the timeout as part of the loop